packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      todo['id'] = createResult1.data[createResultSetName].id;
+      todo['owner'] = userName1;
+      // protected fields are nullified in mutation responses
+      expectNullFields(createResult1.data[createResultSetName], ['privateContent', 'ownersContent']);
+
+      const publicFieldSet = `
+        id
+        publicContent
+      `;
+      // owner cannot update a record with public protected field
+      await expect(
+        async () => await user1TodoHelper.update(updateResultSetName, { id: todo['id'], publicContent: 'Public Content' }, publicFieldSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // owner cannot update a record with a protected field that does not allow update operation
+      await expect(
+        async () => await user1TodoHelper.update(updateResultSetName, { ...todo, ownersContent: 'Owners Content' }),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // non-owner cannot update a record to re-assign ownership
+      const ownerFieldSet = `
+        id
+        owner
+      `;
+      await expect(
+        async () => await user2TodoHelper.update(updateResultSetName, { id: todo['id'], owner: 'user2' }, ownerFieldSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // non-owner cannot update a record to re-assign owners in dynamic owners list
+      const ownersFieldSet = `
+        id
+        authors
+      `;
+      await expect(
+        async () => await user2TodoHelper.update(updateResultSetName, { id: todo['id'], authors: ['user2'] }, ownersFieldSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // non-owner cannot update a record with an owner protected field
+      const ownerContentFieldSet = `
+        id
+        ownerContent
+      `;
+      await expect(
+        async () => await user2TodoHelper.update(updateResultSetName, { ...todo, ownerContent: 'Owner Content' }, ownerContentFieldSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // owner cannot read a record with public protected field
+      const getResult1 = await user1TodoHelper.get({ id: todo['id'] }, publicFieldSet, false, 'all');
+      checkOperationResult(
+        getResult1,
+        { id: todo['id'], publicContent: null },
+        `get${modelName}`,
+        false,
+        expectedFieldErrors(['publicContent'], modelName),
+      );
+
+      const listTodosResult1 = await user1TodoHelper.list({}, publicFieldSet, `list${modelName}`, false, 'all');
+      checkListItemExistence(listTodosResult1, `list${modelName}`, todo['id'], true);
+      checkListResponseErrors(listTodosResult1, expectedFieldErrors(['publicContent'], modelName, false));
+
+      /*
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket